### ISSUE TITLE
Configuration tree must be significantly more clear and more intuitive

### ISSUE THREAD

**Issue by markubiak** [lutris/lutris#184]
I really like the config structure of Lutris, once viewing the configuration and documentation, it is very clear the "Game", "Runner", "System" hierarchy.  It is intuitive from a programming standpoint, but not from an interface standpoint.  For example, it is incredibly unclear that, when setting up a game, that the runner configuration and system configuration were inherited from previous configuration.  I ended up manually going through and configuring each game individually.  It was extraordinarily unclear that the inheritance system even existed.

A complete UI redesign is entirely unnecessary; I propose a system that makes inheritence clear.  This comes in two main steps:
*  Renaming the configuration box headings.  The system configuration box should remain unchanged.  The headings of the "Runner" type configuration box (found by navigating to Lutris -> Manage Runners -> Configure) should have the headings changed from ("Runner Configuration", "System Configuration") to ("Runner Configuration", "System Overrides").  Similarly, the Game configuration headings should be changed from ("Game Info", "Game Configuration", "Runner Configuration", "System Configuration") to ("Game Info", "Game Configuration", "Runner Overrides", "System Overrides").
EDIT: At this point in the post, I actually downloaded the latest git commit, and this is much better (options vs configuration) but it still isn't very clear.  "Overrides" would be a much better term.  Everything after here is analyzed using the latest commit as of May 4 2015
*  Manually enabling overrides.  The use of the "delete" button is intuitive, but I think many will not get this inheritence style.  The simplest way I can think of is a toggle for each pane.  I'll use the example of a game-type configuration window for a native steam game.  The game info and game options panes should be left alone, but the "Runner Overrides" and "System Overrides" sections should have universal checkboxes at the top, entitled "[  ] Enable overrides for the Steam runner" and "[  ] Enable overrides for the system configuration" respectively.  When unchecked, these boxes would leave every option in the pane greyed out, in order to notify the user that nothing will be changed.  When checked, the options would become available just as they currently are (the improvements like the delete button and the "no change" options for comboboxes should stay similar, I can't think of any better way to have a small checkbox return one of (True, False, None).

I hope this makes some sense, and if it doesn't, let me know and I'll try to clarify.  Great application, by the way; I would try to help but most of my concerns are UI-related and I have very little GTK knowledge.  I'm relatively competent at basic Python libraries, though, if you have anything I could help with.

**Comment by Xodetaetl** [comment]
Thanks for this analysis ! I fully agree that the UI currently doesn't expose the cascading structure in an explicit, intuitive way and I like your propositions, it sounds like the best way to go.

**Comment by markubiak** [comment]
Excellent, thank you very much!  I'll look at icon customization, I already figured it out somewhat by adding banners and icons manually.

**Comment by Xodetaetl** [comment]
> I'll look at icon customization

Great !

I think the place for this option is in the Game Info tab, which is meant to hold the game's metadata (the game's genre, release year, etc.). This tab, this set of game data, is currently out of the "config options system" (config.py), and is held in the sqlite database as opposed to yaml files for config options. We've planned to move everything to the database, but I don't know if then we'll merge the Game Info stuff into the config system, but I think we'll be able to and should.
(And if you're wondering, yes, the runner combobox shouldn't be in the Game Info tab. We need to move it. Perhaps put it back above the tabs, where it was a few versions ago.)

So, here's some guidance:
- You'd add the file browsers for the icons paths in `build_info_tab` from config-dialogs.py
- There's a class to generate the file browsers in widgets.py: `FileChooserEntry`
- To save you'd probably expand the `pga.add_or_update()` part of `Game.save()`
- To update the UI with the new images, there's `GameView.update_row()` from widgets.py

**Comment by markubiak** [comment]
I was thinking on this.  What if under ~/.local/share/lutris I add a custom_icons and custom_banners directory.  The function to build the main view will check in those directories for the icons/banners first, then if nothing's there go to the main ~/.local/share/icons or banners directory.  To get it up and running, I'd have a dialog in the "Game Info" that just copies the selected file to the custom (icons|banners) directory and names it accordingly.  That'll give a base if someone has a 184x69 image or a 32x32 icon already, cropping will be harder.  I'm very familiar with the ImageMagick set of tools from my wallpaper script, but AFAIK there's no way to have it create a crop GUI.

**Comment by Xodetaetl** [comment]
It sounds good to me. Except that icons need to be in `~/.local/share/icons/hicolor/32x32/apps` to get used by desktop shortcuts. You could add a "_custom" suffix to the file names to avoid overriding default icons.
And custom banners could be in a subfolder of the banners folder. (Just having OCD here. :)

**Comment by markubiak** [comment]
So, just to confirm, with Portal as an example:

Icon:
Default: ~/.local/share/icons/hicolor/32x32/lutris_portal.png
Custom: ~/.local/share/icons/hicolor/32x32/lutris_portal_custom.png

Banner:
Default: ~/.local/share/lutris/banners/portal.jpg
Custom: ~/.local/share/lutris/banners/custom/portal.jpg

**Comment by Xodetaetl** [comment]
That's it ! :)

**Comment by markubiak** [comment]
First off, this is a project for me to learn Python classes a little better and to learn some GTK, so this is going to be a little slow.  I had an hour or two, I modified widgets.py to grab custom icons and banners over their regular counterparts, no flaws there.  I added a FileChooserButton on the Game Info tab and have it setup so that it copies the file to the proper location using the banner_file_chooser.connect("file-set", self.on_custom_banner_select) function (I think this is called Gtk callbacks?) .  It isn't ideal, as it will copy the file as soon as the "Open" button is pressed.  I'd prefer to do this on clicking the "edit" button / on_save() function, but I'm not quite sure how.  I'll get a proper git fork up and running tomorrow morning so you can see.

I would like to move this discussion elsewhere, as it doesn't pertain to the original topic.  What would be the best place for this discussion?

**Comment by Xodetaetl** [comment]
We can move the discussion to a pull request of your fork.

**Comment by markubiak** [comment]
[Here's the pull request](https://github.com/lutris/lutris/pull/186)

This is a just a basic update to add support for properly pulling icons; I have a lot more but it's not stable or pretty enough for a PR.

**Comment by Xodetaetl** [comment]
Back to initial topic : 
I implemented the request a bit differently, I added a small informative text instead of the "Allow overrides" checkbox that you proposed. And I didn't add "overrides" to the tab titles, I felt it wasn't necessary and would cause translation headakes or mistakes (there's no translation for "override" as a noun in French for instance).

Let me know if you agree or disagree with the changes.
Meanwhile, I close the issue. :)
